### PR TITLE
Remote CI Builds

### PR DESCRIPTION
**Pull request by abejgonzalez**
**Related issue**: #301 

**Type of change**: other enhancement

**Impact**: no rtl change

**Development Phase**: implementation

**Release Notes**
This allows support for remote CI builds on the millennium machines. It builds verilator and the rtl on the servers and sends the build files back to the CI docker instances to run the tests. Additionally, this PR adds back the Large/Mega configs for testing. The supported tests are below (mainly chosen to reduce runtime).

Mega/Large:
- Only bmark tests
- Csmith added

### CODE REVIEW

**Review by bkorpan — COMMENTED** [review]
Should we tweak the number of csmith iterations so that each of the configs finish at around the same time?

**Review by bkorpan — APPROVED** [review]

**Comment by abejgonzalez** [comment]
@bkorpan So I think that the number will vary alot whenever a config changes significantly. So personally don't want to get into the habit of changing it alot. However, I think it is something that can be changed for this PR.

**Comment by abejgonzalez** [comment]
On another note: I am going to mark this PR as "DO NOT MERGE" for a bit. I want to change it so that there is no 'cleanup' step and 'copy' step. This makes it so that there can be dangling files left on the server. Instead, I want to make it so that each step is fully independent of each other.

**Review by bkorpan — APPROVED** [review]

File: .circleci/defaults.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# local variables (aka within the docker container)
+LOCAL_CHECKOUT_DIR=$HOME/project
+LOCAL_RISCV_DIR=$HOME/riscv-tools-install
+LOCAL_CHIPYARD_DIR=$HOME/chipyard
+LOCAL_VERILATOR_DIR=$HOME/verilator
+LOCAL_SIM_DIR=$LOCAL_CHIPYARD_DIR/sims/verisim
+
+# key value store to get the build strings
+declare -A mapping
```
Comment (jerryz123) [inline .circleci/defaults.sh:35]: What is the point of this?

**Review by jerryz123 — APPROVED** [review]

**Pull request merged**